### PR TITLE
Narrow down routing of Document API GetDocument messages.

### PR DESCRIPTION
**Pull request by toregge**
@geirst: please review

### CODE REVIEW

File: document/src/main/java/com/yahoo/document/select/rule/DocumentNode.java [hunk 1]
```diff
@@ -46,6 +46,9 @@ public Object evaluate(DocumentOperation op) {
         } else if (op instanceof DocumentRemove) {
             DocumentRemove removeOp = (DocumentRemove)op;
             return (removeOp.getId().getDocType().equals(type) ? op : Boolean.FALSE);
+        } else if (op instanceof DocumentGet) {
+            DocumentGet removeOp = (DocumentGet)op;
```
Comment (geirst) [inline document/src/main/java/com/yahoo/document/select/rule/DocumentNode.java]: Rename variable?

**Review by geirst — APPROVED** [review]
👍 Only a nitpick

**Comment by geirst** [comment]
@vekterli please review as well

File: document/src/main/java/com/yahoo/document/DocumentGet.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    private final DocumentId docId;
+
+    public DocumentGet(DocumentId docId) { this.docId = docId; }
+
+    @Override
+    public DocumentId getId() { return docId; }
+
+    @Override
+    public void setCondition(TestAndSetCondition condition) {
+        throw new IllegalStateException("conditional DocumentGet is not supported");
```
Comment (vekterli) [inline document/src/main/java/com/yahoo/document/DocumentGet.java]: Consider `UnsupportedOperationException` instead

**Review by geirst — APPROVED** [review]
👍

**Review by vekterli — APPROVED** [review]
:+1: only a minor comment

**Review by vekterli — APPROVED** [review]

**Pull request merged**